Tasks/UseDotNetV2/models.ts

```diff
@@ -149,16 +149,22 @@ export class VersionParts {
         try {
             let parts = version.split('.');
             // validate version
-            if ((parts.length < 2 || parts.length > 3) || // check if the version has 2 or 3 parts
+            if (parts.length < 2 || // check if the version has at least 3 parts
                 (parts[1] == "x" && parts.length > 2) ||  // a version number like `1.x` must have only major and minor version
                 (parts[1] != "x" && parts.length <= 2) ||  // a version number like `1.1` must have a patch version
                 !parts[0] || // The major version must always be set
                 !parts[1] || // The minor version must always be set
                 (parts.length == 3 && !parts[2]) || // a version number like `1.1.` is invalid because the patch version is missing
                 Number.isNaN(Number.parseInt(parts[0])) || // the major version number must be a number
-                ( // The next lines check the minor version
-                    Number.isNaN(Number.parseInt(parts[1])) &&  // the minor version number must be a number
-                    parts[1] != "x" // the minor version can't be `x`
+                (
+                    parts[1] != "x" && // if the minor version is not `x`
+                    (
+                        Number.isNaN(Number.parseInt(parts[1])) || // the minor version number must be a number
+                        (
+                            parts.length > 2 && parts[2] != "x" && // if the patch is not `x`, then its an explicit version
```

Comment:
Will `ValidateVersionSpec` be called if explicit version is specified?

Reply:
ValidateExplicitVersionNumber will be called if global.json is used
ValidateVersionSpec will be called if global.json is not used, irrespective of whether the version is explicit or not.